### ISSUE TITLE
soft_or_hard checking for 1 instead of true causes soft bounce instead of hard bounce

### ISSUE THREAD

**Issue by lunatyq** [sisimai/rb-Sisimai#83]
Because this call returns true, false or nil:
`getchecked = Sisimai::SMTP::Error.is_permanent(argv2)`

https://github.com/sisimai/rb-Sisimai/blob/d56fc6a0b1a808095eb2793f331bce70f6abbc55/lib/sisimai/smtp/error.rb#L95-L96

Now it is:
```
# The value is 0 or 1
softorhard = getchecked == 1 ? 'hard' : 'soft'
```

Should be:
```
softorhard = getchecked ? 'hard' : 'soft'
```

**Comment by azumakuniyuki** [comment]
Thanks for the report ! The returned value from is_permanent was true, false or nil, I made sure just now.

**Comment by azumakuniyuki** [comment]
This issue has been resolved at pull-request #84
